pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -48,4 +52,66 @@ public void testAddOrUpdateGauge() {
     controllerMetrics.removeGauge(metricName);
     Assert.assertTrue(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
   }
+
+  @Test
+  public void testConcurrentGaugeUpdates() throws InterruptedException {
+    PinotConfiguration pinotConfiguration = new PinotConfiguration();
+    pinotConfiguration.setProperty(CONFIG_OF_METRICS_FACTORY_CLASS_NAME,
+        "org.apache.pinot.plugin.metrics.yammer.YammerMetricsFactory");
+    PinotMetricUtils.init(pinotConfiguration);
+    ControllerMetrics controllerMetrics = new ControllerMetrics(new YammerMetricsRegistry());
+    String metricName = "testConcurrent";
+
+    // update and remove gauge simultaneously
+    ExecutorService service = Executors.newFixedThreadPool(3);
+    IntStream.range(0, 1000).forEach(i -> {
+      controllerMetrics.setOrUpdateGauge(metricName, () -> (long) i);
+    });
```

Comment:
The service threads seems not used while updating the gauge. Maybe you are trying something like:

`service.submit(() -> controllerMetrics.setOrUpdateGauge(metricName, () -> (long) i))`

In that case, it might become difficult to assert the final value of the gauge as the last updated value will be non-deterministic. 
We can just remove the thread-pool for now.

Reply:
Thank you for the review I have removed this thread-pool